packages/ui-core/src/components/Select/Select.tsx

```diff
@@ -106,31 +103,51 @@ class Select extends React.Component<ISelectProps, ISelectState> {
     itemWrapperNode: HTMLDivElement;
     itemsNodeList: HTMLDivElement[];
     selectNode: HTMLDivElement;
-    comboboxNode: HTMLInputElement;
+
     isTouch: boolean = detectTouch();
 
     constructor(props: ISelectProps) {
         super(props);
 
         this.state = {
             isOpen: false,
-            focus: false,
             activeIndex: 0,
             comboboxItems: props.items,
             inputValue: '',
         };
+        this.itemsNodeList = [];
     }
 
-    componentDidMount() {
-        document.addEventListener('click', this.onClickOutside);
+    onEventListener = () => {
+        const { isOpen } = this.state;
+
+        if (!isOpen) {
+            document.addEventListener('click', this.onClickOutside);
+        }
+        if (isOpen) {
+            document.removeEventListener('click', this.onClickOutside);
+        }
     }
 
-    componentWillUnmount() {
-        document.removeEventListener('click', this.onClickOutside);
+    onClickOutside = (e: MouseEvent): void => {
+        const { isOpen } = this.state;
+
+        if (e.target instanceof Node && this.selectNode.contains(e.target) || !isOpen) {
+            return;
+        }
+
+        this.setState({ isOpen: false });
     }
 
     handleClickTitle = (): void => {
```

Comment:
В этом методе и в том, что ниже - копипаста. Лучше
```suggestion
    handleOpenDropdown = (): void => {
```
И использовать его в обоих местах.